src/carp_viz.cpp

```diff
@@ -115,6 +115,10 @@ Rcpp::List CARP_VIZcpp(const Eigen::VectorXd& x,
     //
     //
     while(rep_iter){
+      // Safety check - only so many iterations of inner loop before we move on
+      if(try_iter > ti){
+        rep_iter = false;
```

Comment:
Using a break statement here may be cleaner than checking for rep_iter in every branch below.